packages/story-editor/src/components/panels/design/textStyle/textStyle.js

```diff
@@ -65,10 +69,38 @@ function StylePanel(props) {
     }));
 
   const [fontsFocused, setFontsFocused] = useState(false);
+  const [failedContrast, setFailedContrast] = useState(false);
+  const { selectedElements } = props;
 
   // Update size and position if relevant values have changed.
   usePresubmitHandler(getUpdatedSizeAndPosition, []);
 
+  const { pages, currentPage } = useStory(
+    ({ state: { pages, currentPage } }) => ({
+      pages,
+      currentPage,
+    })
+  );
+  const pageSize = useLayout(({ state: { pageWidth, pageHeight } }) => ({
+    width: pageWidth,
+    height: pageHeight,
+  }));
+
+  useEffect(() => {
+    const selectedElementIds = selectedElements.map(
+      (selectedElement) => selectedElement.id
+    );
+
+    getFailingPages(pages, pageSize, currentPage).then((failedPages) => {
+      // getFailingPages returns an array of pages, since we only care
+      // about currentPage, we can grab the single page result.
+      const result = failedPages[0]?.result;
+      setFailedContrast(
+        result?.some((elementId) => selectedElementIds.includes(elementId))
+      );
+    });
+  }, [currentPage, pageSize, pages, selectedElements]);
```

Comment:
💭 It would be ideal if this effect only ran when `currentPage` changed (like before).

I think we could derive whether the selected text element has a contrast warning rather than setting it here.

```js
  useEffect(() => {
    const selectedElementIds = selectedElements.map(
      (selectedElement) => selectedElement.id
    );

    getFailingPages(pages, pageSize, currentPage).then((failedPages) => {
      // getFailingPages returns an array of pages, since we only care
      // about currentPage, we can grab the single page result.
      const result = failedPages[0]?.result;
      setFailedPages(result); // set failed ids in state
    });
  }, [currentPage, pageSize, pages, selectedElements]);
...

const showContrastWarning = failedPages?.some((elementId) => selectedElementIds.includes(elementId));

return (
    ...
    {showContrastWarning && <Warning message={COLOR_COMBINATION} />}
    ...
)
```